indico/web/templates/forms/datetime_widget.html

```diff
@@ -114,6 +154,12 @@
                 linkedDateField.datepicker().data('onSelectHandlers').push(updateDateLimits);
                 updateDateLimits(linkedDateField.datepicker('getDate'));
             {% endif %}
+
+            resetFieldAction.on('click', function resetDateTime() {
+                timeField.val('');
+                timeField.trigger('change');
```

Comment:
matter of personal preference, but I'd probably chain this into the previous line:

    timeField.val('').trigger('change');

they are kind of related anyway